src/rust/terminusdb-community/src/graphql/mod.rs

```diff
@@ -39,6 +36,7 @@ predicates! {
         let frames: AllFrames = context.deserialize_from_term(&frame_term).expect("Unable to parse frames into rust struct");
         let mut sanitized_frames: AllFrames = frames.sanitize();
         sanitized_frames.invert();
+        sanitized_frames.calculate_subsumption();
```

Comment:
I think we should deserialize frames to an intermediate type, then convert that to our final type to do these finalization steps you have in here. That way it's more clear that these steps only happen once as part of a construction, instead of being methods that can be called at arbitrary times.